### DISCUSSION TITLE
How to import SVG as a svgo component

### DISCUSSION THREAD

**Discussion by kumarasinghe** [jaredpalmer/razzle#1609] [Q&A] [answered]
### Current Behavior

When you try to import a SVG file as a react component following error occurs:

`Invalid tag http://localhost:3001/static/media/logo.935ec323.svg`

### Expected behavior

SVG should be rendered as an embedded component

### Steps to reproduce

`yarn add @svgr/webpack`

```
// razzle.config.js
module.exports = {
  modifyWebpackConfig(opts) {
    const config = opts.webpackConfig;
    config.module.rules.push({
      test: /\.svg$/,
      use: ["@svgr/webpack"],
    });
    return config;
  }
};
```
```
// app.js
import Logo from "logo.svg";

export default class App extends React.Component {
  render() {
    return (
      <div>
          <Logo/>
      </div>
    );
  }
}
```

### Your environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Razzle           | 4.0.4 |
| Razzle Plugins   | razzle-plugin-scss |
| Node             | v14.16.0 |
| Browser          | Chrome 89.0.4389.90 (64-bit) |
| npm/Yarn         | Yarn 1.22.10 |
| Operating System | Ubuntu 20.04 64Bit |
| TypeScript       | No | 
| React            | 17.0.2 |

**Comment by kumarasinghe** [comment]
@jaredpalmer FYI

**Comment by squiddy** [comment]
Can confirm this is happening on a fresh `create-razzle-app` application with your configuration applied.

I think this is caused by the default configuration. There is already a `file-loader` configured for `.svg`. One solution could be this:

```js
  modifyWebpackOptions({ options: { webpackOptions } }) {
    webpackOptions.fileLoaderExclude = [
      /\.svg$/,
      ...webpackOptions.fileLoaderExclude,
    ];
    return webpackOptions;
  },
``` 

However, that prevents you from refereing to SVGs as external assets, e.g. in CSS files. If you follow the configuration over at https://react-svgr.com/docs/webpack/#using-with-url-loader-or-file-loader it works (*together with fileLoaderExclude*).

```js
  modifyWebpackConfig(opts) {
    const config = opts.webpackConfig;
    config.module.rules.push({
      test: /\.svg$/,
      use: ["@svgr/webpack", "url-loader"],
    });
    return config;
  },
```

Usage:

```js
import React from "react";
import logoUrl, { ReactComponent as Logo } from "./react.svg";

class Home extends React.Component {
  render() {
    return (
      <>
        <img src={logoUrl} />
        <Logo />
      </>
    );
  }
}

export default Home;

```

**Comment by squiddy** [comment]
Not sure if this is a bug. I also encountered a similar issue when I introduced specific loaders for SVGs. Maybe this is more of a gotcha that could be documented?

**Comment by fivethreeo** [comment]
Use a resourceQuery and add the rule at the top, just use unshift instead of push, so you can do 

import Logo from "logo.svg?component";

https://webpack.js.org/configuration/module/#ruleresourcequery

**Comment by fivethreeo** [comment, marked as answer]
https://gist.github.com/fivethreeo/1b37aa5bfd99eb3ecdfb7aa6039cab40
